### DISCUSSION TITLE
Saving a Tif with multiple layers

### DISCUSSION THREAD

**Discussion by MCITS-MS** [dlemstra/Magick.NET#680] [Help]
I am trying to load a tif and save it as a tif. 
Unfortunately the layer information is always lost.

I use the latest version of "Magick.NET-Q8-AnyCPU" (7.20.0.1) via NuGet.

I use the following code to load and save the image:
```
using var sourceImage = new MagickImage("Example.tif");
sourceImage.Write("Output.tif");


using var collectionImage = new MagickImageCollection("Example.tif");
collectionImage.Write("Output2.tif");
```

In the zip you can find the original tif with 2 layers as well as the outputs of both tests:

[images.zip](https://github.com/dlemstra/Magick.NET/files/4832023/images.zip)

**Comment by dlemstra** [comment]
What do you mean by layer information?

**Reply by MCITS-MS** [reply]
![image](https://user-images.githubusercontent.com/67427060/85833577-3c1f1680-b792-11ea-9a89-b1672fcc54d7.png)
In Photoshop there are several layers. The layers will be lost.

**Comment by dlemstra** [comment]
That is the expected behavior. This library does not understand how to create Photoshop layers and this means that you will loose that information.

**Reply by MCITS-MS** [reply]
In an older version this works without errors.
If I use the same code with this version, I get a new Tif with the same layers.
![image](https://user-images.githubusercontent.com/67427060/85837513-64aa0f00-b798-11ea-9869-986ae7b455e3.png)

**Comment by dlemstra** [comment]
It turns out that was a bug we fixed because there was a problem with tiff pages and photoshop layers. If you want layers instead of pages you will need to specify that. In the next version you will be able to do this:

```C#
var defines = new TiffWriteDefines()
{
    WriteLayers = true
};
collectionImage.Write("Output2.tif", defines);
```

**Comment by MisterMiros** [comment]
Hi. I'm having the same problem as the topic starter. I'm reading, cropping and the writing the file and all the layers are lost in the process. Even if I disabled the cropping. Here's the code I'm using
```
var settings = new MagickReadSettings()
{
    Compression = CompressionMethod.NoCompression,
    Format = MagickFormat.Tif,
    Verbose = true,
    Debug = true,
    Defines = new TiffReadDefines
    {
        IgnoreLayers = false,
    },
};
using var image = new MagickImageCollection();
image.Read("original.tif", settings);
foreach (var i in image)
{
    i.Crop(390, 590);
}
var writeDefines = new TiffWriteDefines
{
    WriteLayers = true,
};
image.Write("result.tif", writeDefines);
```

**Reply by dlemstra** [reply]
Can you share a file that can be used to reproduce your issue @MisterMiros and next time please start a new discussion where you mention this one as a reference.

**Reply by MisterMiros** [reply]
Yes, here is the [google drive link](https://drive.google.com/file/d/1x1fQmiBv2RDRXnsDGd4C0BZlftI2bw1D/view?usp=share_link.)

**Reply by dlemstra** [reply]
The layers are not lost. The written tiff image contains psd layers. But not all information of the original layer is written to the output image. Supported for writing layers is very limited.
